### ISSUE TITLE
Unable to "find" the app in teams

### ISSUE THREAD

**Issue by EliseLR** [OfficeDev/microsoft-teams-apps-champion-management#47]
Hi,

We have now successfully uploaded the application and synced to Teams. We find the app in the Teams Admin Center under the apps overview. However, when we try adding the app as a tab in a team, we are unable to find the app. We search for "Champion" but no app appears, even when we look in the all apps overview. See screenshot below: 
![image](https://user-images.githubusercontent.com/87803032/126607791-00ffc5a8-aa6e-4d51-b443-e72ebc8fe564.png)

In the Teams Admin Center, in the apps overview, we have also tried clicking the "add to team" button. Still, the app does not appear.

![image](https://user-images.githubusercontent.com/87803032/126608125-1e21ac12-6724-42d6-8c07-23a30eb751f9.png)

Could you please assist? Thank you :-)

PS: we have tried restarting teams and have waited 24 hours since we synced the app to Teams, but it still does not work. 

Regards, Elise.

**Comment by v-saikirang** [comment]
Hi @EliseLR , Thanks for trying this app. We have updated the package to a new version V1.2 yesterday with few fixes and improvements. Can you please uninstall the existing package and try installing the new package from the same location - [link](https://github.com/OfficeDev/microsoft-teams-apps-champion-management/tree/main/sharepoint/solution). 

You can see the updated ReadMe [here](https://github.com/OfficeDev/microsoft-teams-apps-champion-management) for the 1.2 changelog and upgrading instructions as well under the section "**Upgrade to version 1.2.0.0**".

**IMPORTANT NOTE**: Please wait for 15 minutes after you have completed first 4 steps under the section **"Upgrade to version 1.2.0.0"** in ReadMe and then continue with step 5. This is to avoid cache related issues for the deleted assets.

**Comment by EliseLR** [comment]
Hi,
thanks for your input. We deleted the app and installed the new version, but the app still does not appear in Teams. 
The SPO site is created, and here I have added Everyone Except Externals to the members group with edit rights. Even so, the app does not appear when we look for it in Teams.

I find the app in Teams under the "Submit to your organisation's app catalogue" overview: 
![image](https://user-images.githubusercontent.com/87803032/127114962-f66c59de-ca58-4223-9087-60bff6cf486a.png)

Here, as you can see, the app status says approved, which should mean it appears under "Built for your org", but no such luck. 
![image](https://user-images.githubusercontent.com/87803032/127115044-12202eca-5eac-4401-b53d-ae7db84f3d23.png)


Please assist with guidance on how we should proceed to fix this. Thank you :-)

**Comment by EliseLR** [comment]
Are there any settings we need to check in Teams or SPO Admin Center for instance?
Thanks :-)

**Comment by v-saikirang** [comment]
Hi @EliseLR ,

You have mentioned the Share point site is created. This happens only after you have installed the app in the teams. May i know how did you you install the app since you mentioned it's not visible on Teams. And if this is already installed can you also please try adding this as a tab under a Teams. Can you also cross check if all the steps are followed as mentioned under the section "Deploy your Package to SharePoint"  in the ReadMe document. I assume you have also clicked on "Sync to Teams" from the App catalog. 

Thanks.

**Comment by EliseLR** [comment]
Hi,
We never found the app when trying to add it as a tab in Teams, or when trying to add it as a personal app in Teams. We attempted to add the app to a team via the Teams Admin Center, which then might be what triggered the creation of the site.

We have now deleted the app according to the guide, and will try to re-install according to the steps specified in the guide. Will let you know how it turns out :-) 

Thanks so far!

**Comment by v-saikirang** [comment]
Hi @EliseLR , 

Before reinstalling can you also check if there is a Teams App Policy that's preventing custom Apps from Teams Admin portal.

![image](https://user-images.githubusercontent.com/87538202/127122986-8c711374-7073-41e6-ab10-50aecbb8232b.png)


Check the "Custom Apps Section". You may need to unblock if custom apps are blocked here.

Please wait for 15- 20 minutes before re installing your app after the 4 steps to uninstall the app. And do not install the app through Teams Admin center. Follow the instructions under "Deploy your Package to SharePoint" in the ReadMe document.
Thanks.

**Comment by EliseLR** [comment]
Thanks,

We have now updated the Teams App policy to allow this app. We then tried reinstalling, however, we are no unable to "sync to teams". When we try to sync the app from SPO app store we just get the error saying "Sync to teams failed". 

I have contacted our global admins to see if they can try to sync or if there is some approval needed or something.

**Comment by v-saikirang** [comment]
Hi @EliseLR ,

The Teams App policy setting may take some time to take effect. Please try below steps after some time. It's advised to repeat the steps after you have made the policy change.

1) Search for the App in Teams. If you find it delete the App from teams.
2) Verify "ChampionManagementPlatform" site is not under "ActiveSites" or "DeletedSites" in the Sharepoint Admin center. If yes, delete from both. Wait for 15 mins.
3) Delete the package file from App catalog, reupload the package, deploy and "Sync to Teams" from App catalog.
4) Please note the "Sync to teams" should happen only from App catalog.

Please let me know if you see any issues.
Thanks.

**Comment by EliseLR** [comment]
Update: We have successfully synced the app to Teams. My colleague, who I believe has the exact same access as me, has been able to add the app as a tab in Teams. However - I still don't see the app! I am not able to add it in Teams, as I cannot find it, and I do not see the app in the team where my colleague has added it, even though we both are members of the team. 

I have made sure we have access to the Champions Management SPO site which was generated, but still do not see the app in Teams. :-(

**Comment by v-saikirang** [comment]
Hi @EliseLR ,
1) may i know if you have tenant admin rights.
2) can you share a screenshot of the site contents page of the SharePoint site.
3) can you share a screenshot of the contents of the Member list.
4) are you using teams in a browser or client. Can you close and reopen client/browser. If you are using a browser can you try clearing the cache.
Thanks.

**Comment by EliseLR** [comment]
1. I have SPO and Teams Admin access which is enabled via Azure Privileged Identity Manager (PIM). These have both been activated when I have been attempting this, and now as I am searching for the app. 
2. 
![image](https://user-images.githubusercontent.com/87803032/127158823-6e7303ce-5322-4db2-afba-fa224c138141.png)

3. 
![image](https://user-images.githubusercontent.com/87803032/127159171-6cd63f77-690b-459f-a53c-88f7d2583b46.png)

4. I have tried both the desktop client and the browser app but neither works. Have tried a reboot, and I have tried logging out of Teams and back in - nothing works. 


We have now had my colleague adding the app and posting about it in a channel. When I then try to open the app from here I get an error saying I do not have permission to open the app. Which permission does one need to open it? 
![image](https://user-images.githubusercontent.com/87803032/127159621-f9c3065b-c1c7-4381-bb25-06495f24fafb.png)

**Comment by EliseLR** [comment]
I think we figured it out!!
Someone has set some weird pilot policy on my teams user.... Sooo the policy change did not apply to my user... I have updated this now, will give it a while before the changes take place. Will let you know how we get on.

Thanks so much for your help so far!

**Comment by v-saikirang** [comment]
Hi @EliseLR,

From your screenshots I see that the app seems to have installed properly in the back end. So i hope it works for you now. Was there a policy set on your account which is overriding the tenant policy? May i know which policy was it. And can you share a screenshot of the "Member List" in your sharepoint  site. I want to see who is set up as Manager by default for the App. The user with the role "Manager" will have access to the Admin section of the app.
Thanks.

**Comment by EliseLR** [comment]
It now works for us! We had to update the policy, wait a while, and then sign in and out of the teams desktop app. 
I have set my self as admin now :-) 

Thank you so much for your assistance!

**Comment by v-saikirang** [comment]
@EliseLR  Great News! Thanks for working through this with patience. 

It would be great if you can let us know the exact policy that had to change. And i assume this is a user level policy which was overriding the global policy. Please let me know if this is correct.
Thanks.

**Comment by EliseLR** [comment]
We had to change the app policy to allow custom apps. We here decided to just allow the champions app, and block all others, as we currently do not use other custom apps.

What then happened was that my colleague and I for some reason had a different app policy assigned - not the global policy which all users have, but a custom policy used for some sort of pilot. To be honest I am not sure why, but we changed the app policy for our two users to the global app policy. :-) Then after some time and logging out and back in, then it worked :-)
